src/ProjectFileTools.MSBuild/Utilities.cs

```diff
@@ -18,15 +22,20 @@ internal static int GetLine(string text, int position)
                     line++;
                 }
             }
+
             return line;
         }
 
         internal static int GetStartOfLine(string text, int position)
         {
+            Requires.NotNullOrEmpty(text, nameof(text));
+            Requires.Range(position > -1 && position < text.Length, nameof(position), "Position must be positive and less than text.Length");
+
             while (position > 0 && text[position - 1] != '\n')
```

Comment:
I think position can be `<= text.Length` here. Considering you subtract one from it. Is there a reason you went with subtracting 1 from position in the loop?